data/sql_updates/election_outcome.sql

```diff
@@ -1,45 +1,160 @@
--- See discussion in https://github.com/18F/openFEC-web-app/issues/514
+-- This view powers election search
 drop materialized view if exists ofec_election_result_mv_tmp;
 create materialized view ofec_election_result_mv_tmp as
-(
-    select fec_election_yr - 4 as election_yr, *
-    from disclosure.cand_valid_fec_yr
-    where
-        cand_ici = 'I' and
-        cand_office = 'P' and
-        fec_election_yr in (cand_election_yr, cand_election_yr + 4)
-    order by fec_election_yr desc
-)
-union all
-(
-    select distinct on (cand_office_st, cand_election_yr)
-        cand_election_yr - 6 as election_yr, *
+-- get the years with election records by office, state and district
+-- this data starts in 2003
+with election_dates_house_senate as(
+    select
+        election_state,
+        office_sought,
+        election_district,
+        election_yr
+    from fecapp.trc_election
+    where office_sought != 'P'
+        group by
+        election_state,
+        office_sought,
+        election_district,
+        election_yr
+),
+election_dates_president as(
+    select distinct on (election_yr)
+        'US'::char(2) as election_state,
+        office_sought,
+        election_district,
+        election_yr
+    from fecapp.trc_election
+    where office_sought = 'P'
+        group by
+        election_state,
+        office_sought,
+        election_district,
+        election_yr
+),
+-- Pull background info for the races
+-- See discussion in https://github.com/18F/openFEC-web-app/issues/514
+incumbent_info as(
+    select distinct on (cand_office_st, cand_office_district, cand_office, cand_election_yr) *
     from disclosure.cand_valid_fec_yr
     where
-        cand_ici = 'I' and
-        cand_office = 'S'
+        cand_ici = 'I'
     order by
         cand_office_st,
+        cand_office,
+        cand_office_district,
         cand_election_yr desc,
         latest_receipt_dt desc
-)
-union all
-(
-    select distinct on (cand_office_st, cand_office_district, cand_election_yr)
-        cand_election_yr - 2 as election_yr, *
-    from disclosure.cand_valid_fec_yr
+),
+-- This creates election records and the incumbent info if applicable.
+-- We need to do this because a null in the elections date district data
+-- means all the districts in the state but the district is listed,
+-- it only applies to one district.
```

Comment:
Thank you for all of these comments describing what each subset of data represents, it's a huge help! :-)